File: todo-api/src/main/java/org/mermer/todoapi/dto/TaskType.java
package org.mermer.todoapi.dto;

public enum TaskType {
	SOUL("soul"),
	PHYSICAL("physical"),
	KNOWLEDGE("knowledge"),
	SOCIAL("social"),
	ETC("etc"),
	SLEEP("sleep"),
	WASTE("waste");

	private final String value;

	private TaskType(String value){
		this.value = value;
	}

	public String value(){
		return this.value;
	}
}

File: entity_backup/entity/todo_task/task_type.dart
import 'package:flutter/cupertino.dart';
import 'package:times_line/common/common.dart';

enum TaskType {
  soul,
  physical,
  knowledge,
  social,
  etc,
  sleep,
  waste,
  nill,
}

Color getTaskColor(BuildContext context, TaskType taskType ){

  switch(taskType){
    case TaskType.soul:
      return context.appColors.soulColor;
    case TaskType.physical:
      return context.appColors.physicalColor;
    case TaskType.knowledge:
      return context.appColors.knowlegeColor;
    case TaskType.social:
      return context.appColors.socialColor;
    case TaskType.etc:
      return context.appColors.etcColor;
    case TaskType.sleep:
      return context.appColors.sleepColor;
    case TaskType.waste:
      return context.appColors.wasteColor;

    case TaskType.nill:
      return context.appColors.hintText;
  }
}